### ISSUE TITLE
Unable to connect to room with iPhone 7 plus 

### ISSUE THREAD

**Issue by top2tech** [twilio/video-quickstart-swift#118]
run the sample VideoQuickStart, version 1.0.0 from cocoapods
iPhone 6 plus (iOS 10.2.1) can connect to the room successfully,
but iPhone 7 plus(iOS 10.3.1) cannot connect to the room, it tries to connect, after about 30 seconds, returns error: signal connection error.

The samples and my app used to work well with beta7, after I upgraded to version 1.0.0 today, iPhone 7 plus starts to show this problem.

**Comment by ceaglest** [comment]
Hey @top2tech,

We deployed a backend update Tuesday (4/25/2017) that enforces "single point of presence" for participants. This means that if you connect to a Room with the same identity on two different devices you will be disconnected from one device when you connect on the other. We have had some customers who were reusing the same identity for participants report being disconnected. Please be sure you are generating tokens with separate identities.

Thanks,
Chris Eagleston

**Comment by top2tech** [comment]
Thanks Chris.
I tested 1 phone at a time without other participants. iPhone 6 plus can always connect, but iPhone 7 plus always fails.

**Comment by top2tech** [comment]
attached the log below:

2017-04-27 23:05:22.366288-0700 VideoQuickStart[2041:1493454] libMobileGestalt MobileGestaltSupport.m:153: pid 2041 (VideoQuickStart) does not have sandbox access for frZQaeyWLUvLjeuEK43hmg and IS NOT appropriately entitled
2017-04-27 23:05:22.366366-0700 VideoQuickStart[2041:1493454] libMobileGestalt MobileGestalt.c:550: no access to InverseDeviceID (see <rdar://problem/11744455>)
2017-04-27 23:05:22.385218-0700 VideoQuickStart[2041:1493454] [MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
2017-04-27 23:05:22.385753-0700 VideoQuickStart[2041:1493454] [MC] Reading from public effective user settings.
2017-04-27 23:05:22.398103-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Platform]:Supported formats were: (
    "<<TVIVideoFormat: 0x170037860> size: {192, 144}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x170039700> size: {352, 288}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x170039720> size: {480, 360}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x170037600> size: {640, 480}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x170037540> size: {960, 540}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x1700375e0> size: {1280, 720}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x170037680> size: {1920, 1080}, fps: 30, format: 420f>",
    "<<TVIVideoFormat: 0x1700374a0> size: {3088, 2320}, fps: 30, format: 420f>"
)
2017-04-27 23:05:22.400775-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating peer connection signaling thread ...
2017-04-27 23:05:22.401731-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating peer connection worker thread ...
2017-04-27 23:05:22.401811-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating peer connection factory ...
2017-04-27 23:05:22.409529-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Done creating peer connection factory.
2017-04-27 23:05:22.409743-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Adding video source ...
2017-04-27 23:05:22.410040-0700 VideoQuickStart[2041:1493516] DEBUG:TwilioVideo:[Platform]:Start TVIVideoCapturer. <TVICameraCapturer: 0x17005e900>
2017-04-27 23:05:22.410085-0700 VideoQuickStart[2041:1493516] DEBUG:TwilioVideo:[Platform]:Start capturing.
2017-04-27 23:05:22.410128-0700 VideoQuickStart[2041:1493516] DEBUG:TwilioVideo:[Platform]:Prepare for capture.
2017-04-27 23:05:22.410215-0700 VideoQuickStart[2041:1493516] INFO:TwilioVideo:[Platform]:Starting capture pipeline ...
2017-04-27 23:05:22.410826-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating a video track ...
2017-04-27 23:05:22.411141-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:VideoTrackimpl::VideoTrackimpl()
2017-04-27 23:05:22.454641-0700 VideoQuickStart[2041:1493491] DEBUG:TwilioVideo:[Platform]:Adding device input: <AVCaptureDeviceInput: 0x170038220 [Front Camera]>
2017-04-27 23:05:22.460540-0700 VideoQuickStart[2041:1493491] INFO:TwilioVideo:[Platform]:Best device format was: <AVCaptureDeviceFormat: 0x174017af0 'vide'/'420f'  640x 480, { 2- 30 fps}, HRSI:3088x2320, fov:54.201, max zoom:145.00 (upscales @4.82), ISO:23.0-2208.0, SS:0.000013-0.500000, supports wide color>
2017-04-27 23:05:22.957912-0700 VideoQuickStart[2041:1493491] DEBUG:TwilioVideo:[Platform]:Capture start completed
2017-04-27 23:05:22.957987-0700 VideoQuickStart[2041:1493491] INFO:TwilioVideo:[Platform]:... Finished starting capture.
2017-04-27 23:05:22.958027-0700 VideoQuickStart[2041:1493491] INFO:TwilioVideo:[Platform]:Session started running
2017-04-27 23:05:22.959592-0700 VideoQuickStart[2041:1493488] DEBUG:TwilioVideo:[Platform]:Video pipeline will start running
2017-04-27 23:05:50.072532-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Adding audio source ...
2017-04-27 23:05:50.084751-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Adding audio track ...
2017-04-27 23:05:50.085607-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:AudioTrackimpl::AudioTrackimpl()
2017-04-27 23:05:50.087320-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating dedicated notifier queue ...
2017-04-27 23:05:50.087409-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:AsyncIOWorker::AsyncIOWorker()
2017-04-27 23:05:50.089462-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:Timer::Timer(once = true, context = 0x1049cfd28, group = 0)
2017-04-27 23:05:50.089643-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Initiating endpoint configuration refresh
2017-04-27 23:05:50.089769-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:RoomSignalingImpl::RoomSignalingImpl()
2017-04-27 23:05:50.089831-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Initializing SIP signaling stack and worker thread ...
2017-04-27 23:05:50.089437-0700 VideoQuickStart[2041:1493670] INFO:TwilioVideo:[Core]:Starting async I/O worker runloop, using method: kqueue
2017-04-27 23:05:50.093882-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating SIP stack ...
2017-04-27 23:05:50.094463-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Adding IPv4/IPv6 TLS transports ...
2017-04-27 23:05:50.094601-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 0.0.0.0:0 UNKNOWN_TRANSPORT target domain=unspecified mFlowKey=0 ]
2017-04-27 23:05:50.094679-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Creating TLS transport for domain  interface= port=0
2017-04-27 23:05:50.095099-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 0.0.0.0:0 UNKNOWN_TRANSPORT target domain=unspecified mFlowKey=0 ]
2017-04-27 23:05:50.095200-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Creating TLS transport for domain  interface= port=0
2017-04-27 23:05:50.095546-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating SipTU (Transaction User) ...
2017-04-27 23:05:50.095618-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Creating SIP signaling stack worker thread ...
2017-04-27 23:05:50.095697-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:SipSignalingStackSocketServer::SipSignalingStackSocketServer()
2017-04-27 23:05:50.095772-0700 VideoQuickStart[2041:1493454] INFO:TwilioVideo:[Core]:Starting SIP signaling stack worker thread ...
2017-04-27 23:05:50.095923-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:Timer::Timer(once = true, context = 0x1049d04a8, group = 0)
2017-04-27 23:05:50.099018-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Initializing Twilio SIP edge (outbound proxy) ...
2017-04-27 23:05:50.099098-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Querying Twilio SIP edge (endpoint.twilio.com) IP addresses ...
2017-04-27 23:05:50.099259-0700 VideoQuickStart[2041:1493670] INFO:TwilioVideo:[Core]:Received command to add timer, checking ...
2017-04-27 23:05:50.099323-0700 VideoQuickStart[2041:1493670] DEBUG:TwilioVideo:[Core]:Added timer: 0x1742e3b00
2017-04-27 23:05:50.142757-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Total number of IP addresses found: 2
2017-04-27 23:05:50.142853-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Setting Twilio SIP edge (outbound proxy) to: sips:54.172.60.54:5061
2017-04-27 23:05:50.142900-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:DNS resolution was successful
2017-04-27 23:05:50.144802-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Entering SIP signaling stack message processing loop ...
2017-04-27 23:05:50.144972-0700 VideoQuickStart[2041:1493671] DEBUG:TwilioVideo:[Core]:Sending a ping message..
2017-04-27 23:05:50.145318-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Perform SipTU processing ...
2017-04-27 23:05:50.145340-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Queue requests at transport level ...
2017-04-27 23:05:50.145699-0700 VideoQuickStart[2041:1493671] DEBUG:TwilioVideo:[Core]:Socket created in ../resiprocate-1.8/resip/stack/TcpBaseTransport.cxx at line 222, transport: 1
2017-04-27 23:05:50.145892-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Connection::Connection: new connection created to who: [ V4 54.172.60.54:5061 TLS target domain=unspecified mFlowKey=32 ]
2017-04-27 23:05:50.145929-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Creating TLS connection for domain  [ V4 54.172.60.54:5061 TLS target domain=unspecified mFlowKey=0 ] on 32
2017-04-27 23:05:50.145991-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Select for 30000 ms.
2017-04-27 23:05:50.146018-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Process outstanding I/O ...
2017-04-27 23:05:50.146039-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Perform SipTU processing ...
2017-04-27 23:05:50.146053-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Queue requests at transport level ...
2017-04-27 23:05:50.146140-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Select for 30000 ms.
2017-04-27 23:05:50.423028-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:http_client::stateChanged[0 => 1]
2017-04-27 23:05:50.525884-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:Response[Content-Type: application/json]
2017-04-27 23:05:50.526056-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:Response[Date: Fri, 28 Apr 2017 06:05:50 GMT]
2017-04-27 23:05:50.526150-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:Response[Content-Length: 201]
2017-04-27 23:05:50.526239-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:Response[Connection: Close]
2017-04-27 23:05:50.526421-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:http_client::stateChanged[1 => 0]
2017-04-27 23:05:50.526507-0700 VideoQuickStart[2041:1493489] TRACE:TwilioVideo:[Core]:Response: Forbidden, code: 403
2017-04-27 23:05:50.527665-0700 VideoQuickStart[2041:1493489] WARN:TwilioVideo:[Core]:Error retrieving endpoint configuration for video service. Returning default configuration.
2017-04-27 23:05:50.529304-0700 VideoQuickStart[2041:1493489] WARN:TwilioVideo:[Core]:Error accessing endpoint configuration service.
2017-04-27 23:05:50.529922-0700 VideoQuickStart[2041:1493454] DEBUG:TwilioVideo:[Core]:RoomSignalingImpl: State transition successful: kInit -> kConnecting
2017-04-27 23:05:50.530785-0700 VideoQuickStart[2041:1493489] INFO:TwilioVideo:[Core]:Creating peer connection ...
2017-04-27 23:05:50.538133-0700 VideoQuickStart[2041:1493489] INFO:TwilioVideo:[Core]:Adding local stream to peer connection ...
2017-04-27 23:05:50.538846-0700 VideoQuickStart[2041:1493489] DEBUG:TwilioVideo:[Core]:Open -> Updating. Process an event
2017-04-27 23:05:50.538945-0700 VideoQuickStart[2041:1493489] DEBUG:TwilioVideo:[Core]:Create local offer: fc3d32bd-a1a7-a0a7-5996-efdf642b3e33
2017-04-27 23:05:50.542094-0700 VideoQuickStart[2041:1493515] DEBUG:TwilioVideo:[Core]:onCreateSessionLocalDescription fc3d32bd-a1a7-a0a7-5996-efdf642b3e33
2017-04-27 23:05:50.545759-0700 VideoQuickStart[2041:1493515] DEBUG:TwilioVideo:[Core]:PeerConnection state: Updating -> Waiting
2017-04-27 23:05:50.546093-0700 VideoQuickStart[2041:1493489] INFO:TwilioVideo:[Core]:Local SDP is ready
2017-04-27 23:05:50.546775-0700 VideoQuickStart[2041:1493489] INFO:TwilioVideo:[Core]:Queue Description: 1 for PeerConnection: fc3d32bd-a1a7-a0a7-5996-efdf642b3e33.
2017-04-27 23:05:50.547648-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Process outstanding I/O ...
2017-04-27 23:05:50.547772-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Wait was interrupted.
2017-04-27 23:05:50.550057-0700 VideoQuickStart[2041:1493671] INFO:TwilioVideo:[Core]:Adding X-Twilio-AccessToken: eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJqdGkiOiJTS2NkNWFmYmY1MjdhYjJkMGZiODM4NmJlZjhiMDVmNTdlLTE0OTMzNTU2NjQiLCJpc3MiOiJTS2NkNWFmYmY1MjdhYjJkMGZiODM4NmJlZjhiMDVmNTdlIiwic3ViIjoiQUNlMzg4YTc1MTVjMWEzMDEzYmZjNDRhNmZlMTBjODRhMSIsImV4cCI6MTQ5MzM1OTI2NCwiZ3JhbnRzIjp7ImlkZW50aXR5IjoiUmljaGFyZCIsInZpZGVvIjp7ImNvbmZpZ3VyYXRpb25fcHJvZmlsZV9zaWQiOiJWUzEyYzQwY2ZlMmJjN2FiNTE0NzA0MmUyNmU2MzlhMTM1In19fQ.-xEC2A2ccDOgHEfa9IVeuPJlw85LjwpbEmtFUokMheI
2017-04-27 23:05:50.550632-0700 VideoQuickStart[2041:1493671] DEBUG:TwilioVideo:[Core]:
Sending outgoing SIP message
INVITE sip:endpoint.twilio.com;transport=tls SIP/2.0

Via: SIP/2.0/TLS 127.0.0.1;branch=z9hG4bK-524287-1---7ad1b2b6a472ec2f;rport

Max-Forwards: 70

Contact: <sip:bb042e41-e1ce-3062-265f-50f052daf946@127.0.0.1;transport=tls;ob>

To: <sip:orchestrator@endpoint.twilio.com>

From: "bb042e41-e1ce-3062-265f-50f052daf946"<sip:bb042e41-e1ce-3062-265f-50f052daf946@endpoint.twilio.com>;tag=13692824

Call-ID: 1HH1eYwidue7rZ1qhT-Bjg..

CSeq: 1 INVITE

Session-Expires: 120

Allow: INVITE, ACK, CANCEL, OPTIONS, BYE, UPDATE, NOTIFY

Content-Type: application/room-signaling+json

Supported: timer, outbound, path, gruu, room-signaling

User-Agent: TwilioVideo SDK

X-Twilio-AccessToken: eyJjdHkiOiJ0d2lsaW8tZnBhO3Y9MSIsInR5cCI6IkpXVCIsImFsZyI6IkhTMjU2In0.eyJqdGkiOiJTS2NkNWFmYmY1MjdhYjJkMGZiODM4NmJlZjhiMDVmNTdlLTE0OTMzNTU2NjQiLCJpc3MiOiJTS2NkNWFmYmY1MjdhYjJkMGZiODM4NmJlZjhiMDVmNTdlIiwic3ViIjoiQUNlMzg4YTc1MTVjMWEzMDEzYmZjNDRhNmZlMTBjODRhMSIsImV4cCI6MTQ5MzM1OTI2NCwiZ3JhbnRzIjp7ImlkZW50aXR5IjoiUmljaGFyZCIsInZpZGVvIjp7ImNvbmZpZ3VyYXRpb25fcHJvZmlsZV9zaWQiOiJWUzEyYzQwY2ZlMmJjN2FiNTE0NzA0MmUyNmU2MzlhMTM1In19fQ.-xEC2A2ccDOgHEfa9IVeuPJlw85LjwpbEmtFUokMheI

Content-Length: 3442



{"format":"planb","name":"room1","participant":{"revision":1,"tracks":[{"enabled":true,"id":"fa330599a739ab88aee1872e37ff827925c5","kind":"audio"},{"enabled":true,"id":"8e05f0515d1cad1f64759f50167464e92429","kind":"video"}]},"peer_connections":[{"description":{"revision":1,"sdp":"v=0\r\no=- 119334996199684406 2 IN IP4 127.0.0.1\r\ns=-\r\nt=0 0\r\na=group:BUNDLE audio video\r\na=msid-semantic: WMS d3cf97beb9cf13e791da24e6b279de0f7094\r\nm=audio 9 UDP/TLS/RTP/SAVPF 111 103 104 9 0 8 106 105 13 110 112 113 126\r\nc=IN IP4 0.0.0.0\r\na=rtcp:9 IN IP4 0.0.0.0\r\na=ice-ufrag:6Z/M\r\na=ice-pwd:foHcu1J9ZhpJnD3FQmH3EK3e\r\na=fingerprint:sha-256 F5:2B:63:03:DB:91:C8:BA:7E:25:B0:19:34:92:E9:82:76:9B:32:EC:BC:B6:6A:6E:6E:96:74:37:6B:11:6E:93\r\na=setup:actpass\r\na=mid:audio\r\na=extmap:1 urn:ietf:params:rtp-hdrext:ssrc-audio-level\r\na=sendrecv\r\na=rtcp-mux\r\na=rtpmap:111 opus/48000/2\r\na=rtcp-fb:111 transport-cc\r\na=fmtp:111 minptime=10;useinbandfec=1\r\na=rtpmap:103 ISAC/16000\r\na=rtpmap:104 ISAC/32000\r\na=rtpmap:9 G722/8000\r\na=rtpmap:0 PCMU/8000\r\na=rtpmap:8 PCMA/8000\r\na=rtpmap:106 CN/32000\r\na=rtpmap:105 CN/16000\r\na=rtpmap:13 CN/8000\r\na=rtpmap:110 telephone-event/48000\r\na=rtpmap:112 telephone-event/32000\r\na=rtpmap:113 telephone-event/16000\r\na=rtpmap:126 telephone-event/8000\r\na=ssrc:956955274 cname:gg57E8Rl4TtaU6Wp\r\na=ssrc:956955274 msid:d3cf97beb9cf13e791da24e6b279de0f7094 fa330599a739ab88aee1872e37ff827925c5\r\na=ssrc:956955274 mslabel:d3cf97beb9cf13e791da24e6b279de0f7094\r\na=ssrc:956955274 label:fa330599a739ab88aee1872e37ff827925c5\r\nm=video 9 UDP/TLS/RTP/SAVPF 96 98 100 102 97 99 101\r\nc=IN IP4 0.0.0.0\r\na=rtcp:9 IN IP4 0.0.0.0\r\na=ice-ufrag:6Z/M\r\na=ice-pwd:foHcu1J9ZhpJnD3FQmH3EK3e\r\na=fingerprint:sha-256 F5:2B:63:03:DB:91:C8:BA:7E:25:B0:19:34:92:E9:82:76:9B:32:EC:BC:B6:6A:6E:6E:96:74:37:6B:11:6E:93\r\na=setup:actpass\r\na=mid:video\r\na=extmap:2 urn:ietf:params:rtp-hdrext:toffset\r\na=extmap:3 http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time\r\na=extmap:4 urn:3gpp:video-orientation\r\na=extmap:5 http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01\r\na=extmap:6 http://www.webrtc.org/experiments/rtp-hdrext/playout-delay\r\na=sendrecv\r\na=rtcp-mux\r\na=rtcp-rsize\r\na=rtpmap:96 VP8/90000\r\na=rtcp-fb:96 ccm fir\r\na=rtcp-fb:96 nack\r\na=rtcp-fb:96 nack pli\r\na=rtcp-fb:96 goog-remb\r\na=rtcp-fb:96 transport-cc\r\na=rtpmap:98 VP9/90000\r\na=rtcp-fb:98 ccm fir\r\na=rtcp-fb:98 nack\r\na=rtcp-fb:98 nack pli\r\na=rtcp-fb:98 goog-remb\r\na=rtcp-fb:98 transport-cc\r\na=rtpmap:100 red/90000\r\na=rtpmap:102 ulpfec/90000\r\na=rtpmap:97 rtx/90000\r\na=fmtp:97 apt=96\r\na=rtpmap:99 rtx/90000\r\na=fmtp:99 apt=98\r\na=rtpmap:101 rtx/90000\r\na=fmtp:101 apt=100\r\na=ssrc-group:FID 3058988910 1105100747\r\na=ssrc:3058988910 cname:gg57E8Rl4TtaU6Wp\r\na=ssrc:3058988910 msid:d3cf97beb9cf13e791da24e6b279de0f7094 8e05f0515d1cad1f64759f50167464e92429\r\na=ssrc:3058988910 mslabel:d3cf97beb9cf13e791da24e6b279de0f7094\r\na=ssrc:3058988910 label:8e05f0515d1cad1f64759f50167464e92429\r\na=ssrc:1105100747 cname:gg57E8Rl4TtaU6Wp\r\na=ssrc:1105100747 msid:d3cf97beb9cf13e791da24e6b279de0f7094 8e05f0515d1cad1f64759f50167464e92429\r\na=ssrc:1105100747 mslabel:d3cf97beb9cf13e791da24e6b279de0f7094\r\na=ssrc:1105100747 label:8e05f0515d1cad1f64759f50167464e92429\r\n","type":"offer"},"id":"fc3d32bd-a1a7-a0a7-5996-efdf642b3e33"}],"type":"connect","version":1}
2017-04-27 23:05:50.562330-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Perform SipTU processing ...
2017-04-27 23:05:50.562455-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Queue requests at transport level ...
2017-04-27 23:05:50.563079-0700 VideoQuickStart[2041:1493671] WARN:TwilioVideo:[Signaling]:RESIP::TRANSPORT: Can't find matching transport [ V4 127.0.0.1:0 TLS target domain=unspecified mFlowKey=0 ]
2017-04-27 23:05:50.563641-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Select for 29582 ms.
2017-04-27 23:05:50.563987-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Process outstanding I/O ...
2017-04-27 23:05:50.564095-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Perform SipTU processing ...
2017-04-27 23:05:50.564299-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Queue requests at transport level ...
2017-04-27 23:05:50.564620-0700 VideoQuickStart[2041:1493671] TRACE:TwilioVideo:[Core]:Select for 29581 ms.
2017-04-27 23:05:53.104668-0700 VideoQuickStart[2041:1493670] DEBUG:TwilioVideo:[Core]:Timer 0x1049cba10 has been cancelled, removing ...

**Comment by top2tech** [comment]
For the failed log, looks like it tries to connect 127.0.0.1 server?

Via: SIP/2.0/TLS 127.0.0.1;branch=z9hG4bK-524287-1---773b0b8b7ecd362b;rport

**Comment by ceaglest** [comment]
Hey @top2tech,

Thank you for the logs, this is very interesting...

As you said the iPhone 6 Plus has no problems whatsoever connecting. However, on the iPhone 7 Plus it only gets part of the way there.

1. DNS resolution on `endpoint.twilio.com` succeeds, and an IPv4 address is returned.
2. In parallel the HTTP request which fetches endpoint configuration succeeds.
3. The client attempts to establish a TLS connection to `endpoint.twilio.com` based upon the DNS results from #1.

However, this is where things go wrong because the TLS handshake never starts. This indicates to me that the bytes written to the socket aren't ever reaching `endpoint.twilio.com` on port 5061. Eventually the request times out, and raises a signaling error.

What we want to see is:
```
INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: TLS handshake starting (client mode)
INFO:TwilioVideo:[Signaling]:RESIP::TRANSPORT: TLS connected
```

So, this leads me to a few follow up questions:

1. Are the iPhone 6 Plus and the iPhone 7 Plus on the same Wifi and Cellular networks?

2. Is the iPhone 7 Plus behind a firewall that blocks TCP traffic on port 5061? You'll want to make sure that you whitelist our [signaling servers](https://www.twilio.com/docs/api/video/ip-address-whitelisting), in order to connect.

3. If the iPhone 7 Plus is on both Wifi and Cellular connections can you try each one separately?

4. What cellular carrier is your iPhone 7 Plus with? Our QA team may have access to the same model to test against.

Best,
Chris Eagleston

**Comment by top2tech** [comment]
without change anything, I tried again just now, the iPhone7 can connect to a room and video chatted with the iphone6!, on both WIFI and Cellular (T-Mobile).

When I got a repro of this issue in the morning, both of the two phones should be on WIFI.

**Comment by top2tech** [comment]
Not sure if it's good or bad news, as dev/test are usually troubled most by intermittent issue  :)

**Comment by ceaglest** [comment]
Hi @top2tech,

I'm glad to hear you were able to connect to a Room on your iPhone 7 Plus! Nevertheless, we would still like to reproduce the problem if possible. We've got access to an iPhone 7 Plus with a T-Mobile SIM for testing, so I'll let you know if we have any success reproducing the connection failures.

Best,
Chris

**Comment by top2tech** [comment]
I got the repro like 20% of the time on another iphone 6s WIFI.
So I added a retry if connecting room failed. It works just with 1 retry when connecting fails.

Attached the log. If you search "retry", you can see when the retry happens.
[retry-connecting-room.txt](https://github.com/twilio/video-quickstart-swift/files/984141/retry-connecting-room.txt)

**Comment by ptankTwilio** [comment]
@top2tech I tried on iPhone 7 Plus to reproduce the problem in following combinations, however, I could not reproduce the problem.
Carrier x Network x Room Condition  = {Verizon, T-Mobile} x {LTE, wifi-A, wifi-B, wifi-C} x {Empty Room, Room with one participant}

We have noticed that some firewalls have the strict policy where traffic to port 5061 is blocked and most likely this is what you are observing also. To eliminate this problem, we are working on using port 443 for signaling traffic. We are planning to release this soon and I will let you know when we release it.

**Comment by ceaglest** [comment]
Hi @top2tech,

I'm sorry that you've been experiencing connectivity issues with our SDK. As @ptankTwilio mentioned we've been working on supporting signaling traffic through TLS/443 to increase connection reliability in our mobile SDKs.

We just released 1.1.0 of Twilio Video for iOS which uses port 443 for all signaling traffic. If you wouldn't mind giving it a try we would be very interested to see if it resolves your connectivity issues.

https://www.twilio.com/docs/api/video/changelogs/ios#110-june-27-2017

Best,
Chris Eagleston

**Comment by ceaglest** [comment]
Please let us know if you get a chance to pickup one of our newer releases supporting port 443. Until then, I'm going to close out this issue.
